### ISSUE TITLE
[🐛] Android Target targetSdkVersion 31 targetSdkVersion 31 Build Error

### ISSUE THREAD

**Issue by jihokim2** [react-native-image-picker/react-native-image-picker#2018]
### Description
At Android Target targetSdkVersion 31 and targetSdkVersion 31 Build Error Occred

Log is Like this.

node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/ImagePickerModule.java:135: error: cannot find symbol
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
                                                       ^
  symbol:   variable TIRAMISU
  location: class VERSION_CODES
/Users/jhkim/v3-albam-owner/node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/ImagePickerModule.java:143: error: cannot find symbol
            libraryIntent = new Intent(MediaStore.ACTION_PICK_IMAGES);
                                                 ^
  symbol:   variable ACTION_PICK_IMAGES
  location: class MediaStore

node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/ImagePickerModule.java:147: error: cannot find symbol
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
                                                           ^
  symbol:   variable TIRAMISU
  location: class VERSION_CODES

node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/ImagePickerModule.java:152: error: cannot find symbol
                    if (selectionLimit == 0) maxNum = MediaStore.getPickImagesMaxLimit();
                                                                ^
  symbol:   method getPickImagesMaxLimit()
  location: class MediaStore

/node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/ImagePickerModule.java:153: error: cannot find symbol
                    libraryIntent.putExtra(MediaStore.EXTRA_PICK_IMAGES_MAX, maxNum);
                                                     ^
  symbol:   variable EXTRA_PICK_IMAGES_MAX
  location: class MediaStore

node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/ImagePickerModule.java:162: error: cannot find symbol
        } else if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
                                                              ^
  symbol:   variable TIRAMISU
  location: class VERSION_CODES

**Comment by bachbonglong** [comment]
Same Issue .

**Comment by zarkoden** [comment]
same issue

**Comment by pIyUsH00rAj** [comment]
same issue

**Comment by thlindustries** [comment]
Same issue, i've tried to change my sdk version to 28, 29, 30, 31, 32, 33 & tiramisu but none of then solved my problem

**Comment by bonnmh** [comment]
i have same problem 
`Module: 'react-native-image-picker' platform 'android-33' not found.`

**Comment by thlindustries** [comment]
A possible workaround is resetting your react-native-image-picker version, in my app it was ^4.4.0, I fixed it into 4.4.0 and it works... (You need to remove node_modules and yarn.lock then yarn install again)

This bug only happens for Android, using iOS it works fine

The down side is that the lib wont have latest features.... but it'll let you to keep developing

**Comment by bonnmh** [comment]
> 

I am using it fine on version ^4.6.0, but after updating node_modules again, the above error appears when I try to launch android

**Comment by thiagodsantos** [comment]
@vincent-paing can you help us?

**Comment by thiagodsantos** [comment]
@vincent-paing please, can you help us?

**Comment by vincent-paing** [comment]
@thiagodsantos ah my bad, I forgot to update the sdk version with the stable one. Will submit a PR now

**Comment by thiagodsantos** [comment]
@vincent-paing it's ok, thanks! 😃

**Comment by vincent-paing** [comment]
I just looked into the code I could make a project with both `compileSdk` 33 and `31. Could you share me a reproducible repo?

**Comment by jihokim2** [comment]
@vincent-paing My gradle version is 7.4.2. And Android gradle plugin is 7.1.3

**Comment by kesha-antonov** [comment]
> I just looked into the code I could make a project with both `compileSdk` 33 and 31. Could you share me a reproducible repo? Would also like to know the version of gradle and Android gradle plugin you are using. Might be below 7 are not downloading necessary sdk automatically.

I have crash too

This is `build.gradle`
```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "31.0.0"
        minSdkVersion = 21
        compileSdkVersion = 31
        targetSdkVersion = 31
        if (System.properties['os.arch'] == "aarch64") {
            // For M1 Users we need to use the NDK 24 which added support for aarch64
            ndkVersion = "24.0.8215888"
        } else {
            // Otherwise we default to the side-by-side NDK version from AGP.
            ndkVersion = "21.4.7075529"
        }
        supportLibVersion = "31.0.0"
        googlePlayServicesVersion = "17.0.0"
        googlePlayServicesVisionVersion = "19.0.0"
        kotlinVersion = '1.6.10'
        firebaseVersion = '21.1.0' // https://github.com/wix/react-native-notifications/blob/master/lib/android/app/build.gradle
        firebaseMessagingVersion = '21.1.0' // https://github.com/zo0r/react-native-push-notification/blob/master/android/build.gradle
        gradleVersion = '7.1.1'
        buildGradlePluginVersion = '7.0.4'
    }
    repositories {
        google()
        mavenCentral()
        jcenter()
    }
    dependencies {
        classpath("com.android.tools.build:gradle:$gradleVersion") {
            force = true
        }
        classpath("com.facebook.react:react-native-gradle-plugin")
        classpath("de.undercouch:gradle-download-task:5.0.1")
        classpath('com.google.gms:google-services:4.3.8')
        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
    configurations.classpath {
        resolutionStrategy {
            force "com.android.tools.build:gradle:$gradleVersion"
        }
    }
}

allprojects {
    repositories {
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(new File(["node", "--print", "require.resolve('react-native/package.json')"].execute().text.trim(), "../android"))
        }
        maven {
            // Android JSC is installed from npm
            url(new File(["node", "--print", "require.resolve('jsc-android/package.json')"].execute().text.trim(), "../dist"))
        }
        mavenCentral {
            // We don't want to fetch react-native from Maven Central as there are
            // older versions over there.
            content {
                excludeGroup "com.facebook.react"
            }
        }

        google()
        jcenter()
        maven { url 'https://www.jitpack.io' }
    }
}
subprojects {
    afterEvaluate {project ->
        if (project.hasProperty("android")) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion
            }
        }
    }
}
```

**Comment by rush86999** [comment]
same here crash. would love to downgrade a stable version. Anybody know which one?

**Comment by jihokim2** [comment]
@rush86999 4.8.5 is stable

**Comment by rush86999** [comment]
@jihokim2 thanks!

**Comment by ekamnh** [comment]
Hello

i am also getting same error here

can anybody help to resolve this

<img width="1423" alt="image" src="https://user-images.githubusercontent.com/28934054/186594117-540ac774-7523-4aa4-9754-e9220e34f2aa.png">

<img width="667" alt="image" src="https://user-images.githubusercontent.com/28934054/186594146-b0583797-e7c7-45f0-8f57-13344b36c6fe.png">

**Comment by mbyakow** [comment]
> @rush86999 4.8.5 is stable

Downgrade to that version solved problem for me, thanks!

**Comment by ekamnh** [comment]
Hi @rush86999 @mbyakow 

i have tried 4.8.5 version too
<img width="534" alt="image" src="https://user-ima. es.githubusercontent.com/28934054/186602824-4e90536b-f7bd-4c0f-922f-d4041c83644d.png">

giving same error

i am on M1 Pro Mac

please advice

Thanks,
ekam

**Comment by jihokim2** [comment]
@ekamnh react-native-image-picker 4.8.5 not ^4.8.5

**Comment by thlindustries** [comment]
@ekamnh  try to setting the version to 4.0.0 and removing   "^" from package.json, also remove node_modules and yarn.lock and run yarn install again

**Comment by Johan-dutoit** [comment]
The example app runs just fine with 31 set for both targetSdkVersion and compileSdkVersion.

**Comment by ekamnh** [comment]
Thanks @jihokim2 

Your Solution works

**Comment by brax10ward** [comment]
> The example app runs just fine with 31 set for both targetSdkVersion and compileSdkVersion (had to add the `exported` flag though).
> 
> Please share a repro with the example app.

@Johan-dutoit Can you clarify what you mean by adding the `exported` flag?

**Comment by Johan-dutoit** [comment]
> > The example app runs just fine with 31 set for both targetSdkVersion and compileSdkVersion (had to add the `exported` flag though).
> > 
> > Please share a repro with the example app.
> 
> @Johan-dutoit Can you clarify what you mean by adding the `exported` flag?

I believe from Android 12 onwards, you have to specify the whether or not your activity, etc. Is exported within your manifest file.

**Comment by vincent-paing** [comment]
@kesha-antonov could you share a full reproducible project? I can't really help any further because both the example project and a fresh project works correctly for me.

**Comment by jihokim2** [comment]
![AlbamOwner – ~:Library:Android:sdk:sources:android-31:android:os:Build java 2022-08-26 19-02-46](https://user-images.githubusercontent.com/18278545/186880420-c293050c-73d3-4101-999b-1ce6043ca29f.png)

Did you see sdk/source/version? android-31 does not have TIRAMISU.

**Comment by vincent-paing** [comment]
@jihokim2 Agreed, you'd need to target android 13 and compile with sdk 33 to use the latest version since the library uses target sdk 33. If this is not something that can be done to your project, pinning your version to `4.8.5` should be ready for you.

However, after changing to 33 and still not work, I'd be happy to take a look at a reproducible repo to debug further.

**Comment by Johan-dutoit** [comment]
Wonder if it's worth reverting back to 
```
targetSdkVersion rootProject.ext.targetSdkVersion
```

**Comment by vincent-paing** [comment]
Just to make the build work, we could hard code 33 integer instead of using the constants, then the consumer would be able to override the `targetSdkVersion` as they wish

**Comment by nomi9995** [comment]
the same issue is happening on my side

**Comment by umangloria** [comment]
Here is an update, I also faced the same issue that is Module: `'react-native-image-picker' platform 'android-33' not found.`
I am finally able to fix the error on my system by adding Android API 33 SDK platforms from SDK Manager.
After adding just resync and it will be successful.
<img width="986" alt="Screenshot 2022-09-19 at 8 52 46 AM" src="https://user-images.githubusercontent.com/22933346/190945026-40c91598-19bc-4ba0-8185-42b2168bd620.png">

**Comment by ZanaNabaz** [comment]
> > @rush86999 4.8.5 is stable
> 
> Downgrade to that version solved problem for me, thanks!


This worked in my case, thanks!

**Note: for me this error happend after I updated cmake from 3.10.2  to 3.18. Dont know if this is related or not**

**Comment by denialfarizi1** [comment]
i change version to 4.8.5 or 4.8.4

> npm install react-native-image-picker@4.8.5 
or
> npm install react-native-image-picker@4.8.4

**Comment by michelazzam** [comment]
same issue

**Comment by Zakyyy** [comment]
same issue

**Comment by umangloria** [comment]
I was able to fix the issue.
Here is what I did to get 4.10.0 working :
in project build.gradle need to add following at the end of repositories:
       ```
 jcenter() {
            content {
                includeModule("com.yqritc", "android-scalablevideoview")
            }
        }
```

**Comment by umangloria** [comment]
> same issue

Have you tried this ? 
https://github.com/react-native-image-picker/react-native-image-picker/issues/2018#issuecomment-1250526358

**Comment by umangloria** [comment]
> same issue
Have you tried this ?
https://github.com/react-native-image-picker/react-native-image-picker/issues/2018#issuecomment-1250526358

**Comment by eueddem** [comment]
@umangloria your suggestion does not work for me.
@vincent-paing so... version above `4.8.5` does not work with `compiledSdkVersion` < 33, right? I think it's a little bit dangerous, as Google requested to [use target SDK >= 31 as of November 2022](https://developer.android.com/google/play/requirements/target-sdk). I bet most apps are under 32 yet.

**Comment by gnprice** [comment]
One thing that's worth emphasizing is that `compileSdkVersion` is a very different thing from `targetSdkVersion`.

This issue only applies when `compileSdkVersion` is below 33 -- `targetSdkVersion` can be much lower and things are fine.
* -> So the title of the issue should be edited to say `compileSdkVersion`, not `targetSdkVersion`.

Then the important difference between the two is described in this blog post from the Android team at Google:
https://medium.com/androiddevelopers/picking-your-compilesdkversion-minsdkversion-targetsdkversion-a098a0341ebd
Quoting from there:
> It should be emphasized that **changing your compileSdkVersion does not change runtime behavior**. While new compiler warnings/errors may be present when changing your compileSdkVersion, your compileSdkVersion is not included in your APK: it is purely used at compile time. (You should really fix those warnings though — they were added for a reason!)
>
> Therefore it is strongly recommended that you **always compile with the latest SDK**.

In short: the best way to deal with this issue is to simply increase your app's `compileSdkVersion` to 33, the latest. It does not affect runtime behavior -- unlike `targetSdkVersion` -- so increasing it is quite a lightweight change.

**Comment by og-abdul-mateen** [comment]
> Here is an update, I also faced the same issue that is Module: `'react-native-image-picker' platform 'android-33' not found.` I am finally able to fix the error on my system by adding Android API 33 SDK platforms from SDK Manager. After adding just resync and it will be successful. <img alt="Screenshot 2022-09-19 at 8 52 46 AM" width="986" src="https://user-images.githubusercontent.com/22933346/190945026-40c91598-19bc-4ba0-8185-42b2168bd620.png">

This does not work for me. V strange.

**Comment by kalous** [comment]
Hello @vincent-paing Is there any chance to change compileSdkVersion/targetSdkVersion to
rootProject.ext.compileSdkVersion/rootProject.ext.targetSdkVersion as it was before? It is quite complicated to change sdk in our corporate environment.

**Comment by eikhunter** [comment]
This issue is still occurring using `compileSdkVersion = 33` && `targetSdkVersion = 33`

**Comment by saintyusuf** [comment]
This errror still accurs on android. Which version do you guys recommend for early 2025? I saw that people recommend 4.8.5 but I thought it might be too low for nowadays.
